services/datamanager/builtin/builtin.go

```diff
@@ -695,64 +724,68 @@ func isOffline() bool {
 	return err != nil
 }
 
-func (svc *builtIn) sync() {
+func (svc *builtIn) sync(ctx context.Context) {
 	svc.flushCollectors()
-
+	// Lock while retrieving any values that could be changed during reconfiguration of the data
+	// manager.
 	svc.lock.Lock()
-	if svc.syncer != nil {
-		toSync := getAllFilesToSync(svc.captureDir, svc.fileLastModifiedMillis)
-		for _, ap := range svc.additionalSyncPaths {
-			toSync = append(toSync, getAllFilesToSync(ap, svc.fileLastModifiedMillis)...)
-		}
-		syncer := svc.syncer
-		stopAfter := time.Now().Add(time.Duration(svc.syncIntervalMins * float64(time.Minute)))
-		svc.lock.Unlock()
-
-		// Only log if there are a large number of files to sync
-		if len(toSync) > minNumFiles {
-			svc.logger.Infof("Starting sync of %d files", len(toSync))
-		}
-		for _, p := range toSync {
-			syncer.SyncFile(p, stopAfter)
-		}
-	} else {
+	captureDir := svc.captureDir
+	fileLastModifiedMillis := svc.fileLastModifiedMillis
+	additionalSyncPaths := svc.additionalSyncPaths
+	if svc.syncer == nil {
 		svc.lock.Unlock()
+		return
 	}
+	syncer := svc.syncer
+	svc.lock.Unlock()
+
+	// Retrieve all files in capture dir and send them to the syncer
+	getAllFilesToSync(ctx, append([]string{captureDir}, additionalSyncPaths...),
+		fileLastModifiedMillis,
+		syncer,
+	)
 }
 
 //nolint:errcheck,nilerr
-func getAllFilesToSync(dir string, lastModifiedMillis int) []string {
-	var filePaths []string
-	_ = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return nil
-		}
-		// Do not sync the files in the corrupted data directory.
-		if info.IsDir() && info.Name() == datasync.FailedDir {
-			return filepath.SkipDir
-		}
-		if info.IsDir() {
+func getAllFilesToSync(ctx context.Context, dirs []string, lastModifiedMillis int, syncer datasync.Manager) {
+	for _, dir := range dirs {
+		_ = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if ctx.Err() != nil {
+				return filepath.SkipAll
+			}
+
+			// TODO: check for context cancellation and stopAfter time passed
```

Comment:
[cleanup] Is this still a TODO?

Reply:
Nope, left over from Devins implementation, will remove.